File: 2_Ruby_Examples/2_Shapes_LSP/square.rb
require 'rectangle'

class Square < Rectangle
	
	def initialize x, y, side_length
		super(x, y, side_length, side_length)
	end
	
	def set_side_lengths horizontal_length, vertical_length
		if horizontal_length != vertical_length
			 throw Exception.new("For squares, both sides have to have equal lengths.")
		end
		super.set_side_lengths(horizontal_length, vertical_length)
	end
	
end

File: 2_Ruby_Examples/2_Shapes_LSP/Square.java
public class Square extends Rectangle {
	
	public Square(int x, int y, int sideLength) {
		super(x, y, sideLength, sideLength);
	}
	
	public void setSideLengths(int horizontalLength, int verticalLength) {
		if (horizontalLength != verticalLength) throw new IlligalArgumentException("For squares, both sides have to have equal lengths.");
		super.setSideLengths(horizontalLength, verticalLength);
	}
	
}